gravitee-apim-console-webui/src/management/api/api-traffic-v4/analytics/components/api-analytics-proxy-filter-bar/api-analytics-proxy-filter-bar.component.ts

```diff
@@ -108,148 +82,106 @@ export class ApiAnalyticsProxyFilterBarComponent implements OnInit {
-      this.emitFilters({ httpStatuses: httpStatuses });
-    });
-
-    this.form.controls.plans.valueChanges.pipe(takeUntilDestroyed(this.destroyRef)).subscribe((plans) => {
-      this.emitFilters({ plans });
-    });
+    this.setupCommonFormSubscriptions();
+    this.setupProxySpecificFormSubscriptions();
+  }
 
-    this.form.controls.applications.valueChanges.pipe(takeUntilDestroyed(this.destroyRef)).subscribe((applications) => {
-      this.emitFilters({ applications });
-    });
+  protected override getAllFilterChips(): FilterChip[] {
+    const filters = this.activeFilters();
+    const commonChips = this.getCommonFilterChips(filters);
+    const proxyChips = this.getProxySpecificFilterChips(filters);
+    return [...commonChips, ...proxyChips];
   }
 
-  applyCustomTimeframe() {
-    const tf = this.form.controls.timeframe.value;
-    const from = tf?.from?.valueOf() ?? null;
-    const to = tf?.to?.valueOf() ?? null;
+  private getProxySpecificFilterChips(filters: ProxyFilters): FilterChip[] {
+    const chips: FilterChip[] = [];
 
-    const currentFilters = this.activeFilters();
-    const updatedFilters = {
-      ...currentFilters,
-      from,
-      to,
-      period: this.customPeriod,
-    };
+    if (filters?.httpStatuses?.length) {
+      filters.httpStatuses.forEach((status) => {
+        const statusOption = this.httpStatuses?.find((opt) => opt.value === status);
+        chips.push({
+          key: FILTER_KEYS.HTTP_STATUSES,
+          value: status,
+          display: statusOption?.label || status,
+        });
+      });
+    }
 
-    this.filtersChange.emit(updatedFilters);
-  }
+    chips.push(...(this.applicationFilterChips() ?? []));
 
-  refreshFilters() {
-    this.refresh.emit();
+    return chips;
   }
 
-  private removeValueFromFilter(currentList: string[] | null, value: string, formControl: FormControl<string[] | null>): void {
-    const filteredList = (currentList || []).filter((item) => item !== value);
-    formControl.setValue(filteredList.length > 0 ? filteredList : null);
+  private createProxySpecificFormControls(): ProxySpecificForm {
+    return {
+      [FILTER_KEYS.HTTP_STATUSES]: this.formBuilder.control<string[] | null>(null),
+      [FILTER_KEYS.APPLICATIONS]: this.formBuilder.control<string[] | null>(null),
+    };
   }
 
-  removeFilter(key: string, value: string) {
-    if (key === 'httpStatuses') {
-      this.removeValueFromFilter(this.form.controls.httpStatuses.value, value, this.form.controls.httpStatuses);
-    } else if (key === 'plans') {
-      this.removeValueFromFilter(this.form.controls.plans.value, value, this.form.controls.plans);
-    } else if (key === 'applications') {
-      this.removeValueFromFilter(this.form.controls.applications.value, value, this.form.controls.applications);
-    }
+  private setupProxySpecificFormSubscriptions(): void {
+    this.form
+      .get(FILTER_KEYS.HTTP_STATUSES)
```

Comment:
```suggestion
    this.form
      .controls.httpStatuses
```
I would recommend to take advantage of typed forms by accessing the named control. 👍